packages/amazon-bedrock/src/bedrock-provider.ts

```diff
@@ -63,42 +88,33 @@ Create an Amazon Bedrock provider instance.
 export function createAmazonBedrock(
   options: AmazonBedrockProviderSettings = {},
 ): AmazonBedrockProvider {
-  const createBedrockRuntimeClient = () =>
-    new BedrockRuntimeClient(
-      options.bedrockOptions ?? {
-        region: loadSetting({
+  const sign = createSigV4SigningFunction({
+    region: options.region,
+    accessKeyId: options.accessKeyId,
+    secretAccessKey: options.secretAccessKey,
+    sessionToken: options.sessionToken,
+  });
+
+  const getBaseUrl = (): string =>
+    withoutTrailingSlash(
+      options.baseURL ??
+        `https://bedrock-runtime.${loadSetting({
           settingValue: options.region,
           settingName: 'region',
           environmentVariableName: 'AWS_REGION',
           description: 'AWS region',
-        }),
-        credentials: {
-          accessKeyId: loadSetting({
-            settingValue: options.accessKeyId,
-            settingName: 'accessKeyId',
-            environmentVariableName: 'AWS_ACCESS_KEY_ID',
-            description: 'AWS access key ID',
-          }),
-          secretAccessKey: loadSetting({
-            settingValue: options.secretAccessKey,
-            settingName: 'secretAccessKey',
-            environmentVariableName: 'AWS_SECRET_ACCESS_KEY',
-            description: 'AWS secret access key',
-          }),
-          sessionToken: loadOptionalSetting({
-            settingValue: options.sessionToken,
-            environmentVariableName: 'AWS_SESSION_TOKEN',
-          }),
-        },
-      },
-    );
+        })}.amazonaws.com`,
+    ) ?? '';
```

Comment:
the `?? ''` could hide errors

Reply:
The pattern in other providers for this case is to hard-code a fallback default base url. Here, the base url would ideally include the `region`, however we can't be sure we have it, so I'll hard-code `https://bedrock-runtime.us-east-1.amazonaws.com`. Let me know if you have something else in mind -- `getBaseUrl` doesn't allow undefined.

Reply:
Ah, maybe it's better to throw if `region` isn't available... I see similar in the vertex provider. Will consider.